### PR TITLE
Set version with git

### PR DESCRIPTION
**Pull request by arbulu89**
In order to get the version of the rolling container, this PR includes a new way of getting the application version. It uses the next order:
1. `VERSION` env variable during the release creation
2. `hack/get_version_from_git.sh` script output
3. Default version

image

PD: I have already checked solutions like [edeliver](https://github.com/edeliver/edeliver), but they look more complicated than the thing we really need. Anyway, if there is any better way of doing so, I'm all open to apply it!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
+  @version "1.1.0"
+  defp get_version do
+    case System.get_env("VERSION") do
+      nil -> get_version_from_git()
+      version -> version
+    end
+  end
+
+  defp get_version_from_git do
+    case File.cwd!() |> Path.join("hack/get_version_from_git.sh") |> System.cmd([]) do
```
Comment (nelsonkopliku) [inline mix.exs:13]: I am wondering whether this is necessary here.
What I mean is that we're setting the `VERSION` env variable at build time by using `get_version_from_git.sh` so, do we really need to do that here? :thinking: 

The alternative I see would be: either at build time the `VERSION` has been set, or we fallback to the default version.

Am I missing something? would that work?

**Review by nelsonkopliku — COMMENTED** [review]
Looks reasonable to me. Just a tiny thought to better understand.
I am also open to learn for better ways of doing it :smile:

File: mix.exs (hunk 1 above)
Comment (arbulu89) [reply mix.exs:13]: Yes, I guess you are right. I just implemented this way to have the way to debug it in my development environment. 
We can delete this case, no issues with it.

Ok, I will remove it for the sake of simplicity

**Review by nelsonkopliku — APPROVED** [review]
LGTM

File: mix.exs (hunk 1 above)
Comment (arbulu89) [reply mix.exs]: Edit: Ok, now I know why I have added it...
It was for the community version. The `VERSION` replacement only affect the suse container, but the community doesn't have such a replacement or Dockerfile edition, and the `docker/build-push-action@v3` job doesn't make it that straightforward

Comment (nelsonkopliku) [reply mix.exs]: Oh I see, is that the reason why e2e tests are failing?

Anyway, no strong opinion from my side. It is just fine like you did. :muscle:

File: mix.exs [hunk 2]
```diff
@@ -1,11 +1,26 @@
 defmodule Trento.MixProject do
   use Mix.Project
 
+  @version "1.1.0"
+  defp get_version do
+    case System.get_env("VERSION") do
```
Comment (fabriziosestito) [inline mix.exs:6]: you dont need a case here, just `System.get_env("VERSION", get_version_from_git())` should work

File: mix.exs (hunk 1 above)
Comment (fabriziosestito) [reply mix.exs:13]: > and the docker/build-push-action@v3 job doesn't make it that straightforward.
ah I see, have to say I am not a big fan of executing shell scripts from elixir but if there is not another way i am ok with it

File: mix.exs [hunk 3]
```diff
@@ -1,11 +1,26 @@
 defmodule Trento.MixProject do
   use Mix.Project
 
+  @version "1.1.0"
+  defp get_version do
```
Comment (fabriziosestito) [inline mix.exs:5]: `defp` should be at the end of the module by convention

File: mix.exs (hunk 2 above)
Comment (arbulu89) [reply mix.exs]: Thanks! I actually did it this way at some point... I guess too many back and forths hehe

**Pull request merged**